lib/resolver/recursive.js

```diff
@@ -143,6 +143,13 @@ class RecursiveResolver extends DNSResolver {
 
     rc.hops = child.hops;
 
+    if (res.code === codes.SERVFAIL) {
+      rc.bogus = true;
```

Comment:
What's the reason for setting bogus here? SERVFAIL doesn't necessarily mean bogus it could happen for many many reasons. It seems that the code path throws anyway if response code is something other than NOERROR.